### PR TITLE
Create note

### PR DESCRIPTION
**Pull request by amartya-rs**
### In this PR:
- Added create note card from where one can enter note details
- Add title, body, tags, color in note details
- Create a note using API call
- Display the note 
### Preview: https://deploy-preview-3--dopenotes.netlify.app/

### CODE REVIEW

File: src/components/tag-input/TagInput.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import "./tag-input.css";
+
+const TagInput = () => {
+   const { state, dispatch } = useNote();
+
+   //adding a new tag into the tags array
+   const createTag = () => {
+      if (state.tag !== "") {
+         dispatch({ type: "SET_TAGS", payload: state.tag });
+         dispatch({ type: "SET_TAG_INPUT", payload: "" });
```
Comment (iRohitGaur) [inline src/components/tag-input/TagInput.jsx:11]: suggestion: try to use one dispatch to acheive this.

**Review by iRohitGaur — APPROVED** [review]
Nice work! Good to merge with updates 👍🏻

**Review by DivyanshSareen — APPROVED** [review]
The code is clean and easy to read. I would suggest removing the console.log statements in the catch blocks as they might cause errors in the production environment.

File: src/assets/icons.js [hunk 2]
```diff
@@ -157,19 +157,20 @@ const UserIcon = () => {
    );
 };
 
-const CrossIcon = () => {
+const CrossIcon = ({ width, height, handler }) => {
```
Comment (varun13169) [inline src/assets/icons.js]: A better way for more generic usecase,

const CrossIcon = ({ props }) => {

File: src/assets/icons.js [hunk 3]
```diff
@@ -157,19 +157,20 @@ const UserIcon = () => {
...
+         width={width ?? "24"}
+         height={height ?? "24"}
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          className="feather feather-x"
+         onClick={handler}
```
Comment (varun13169) [inline src/assets/icons.js]: use 
{...props}


Similar to 
```
const ColorButton = (props) => {
   return (
      <>
         <button {...props}></button>
      </>
   );
};

export { ColorButton };
```

File: src/assets/icons.js [hunk 4]
```diff
@@ -157,19 +157,20 @@ const UserIcon = () => {
...
 };
 
-const CrossIcon = () => {
+const CrossIcon = ({ width, height, handler }) => {
    return (
       <svg
          xmlns="http://www.w3.org/2000/svg"
-         width="24"
-         height="24"
+         width={width ?? "24"}
```
Comment (varun13169) [inline src/assets/icons.js]: No need of nullish coalescing operator

File: src/components/color-pallete/color-palette.css [hunk 5]
```diff
@@ -0,0 +1,61 @@
+/*color palette styling*/
+.color-palette {
+   width: 7rem;
+   height: max-content;
+   padding: 10px;
```
Comment (varun13169) [inline src/components/color-pallete/color-palette.css:5]: Shouldn't padding change with font size? use rem.

File: src/components/create-note-card/create-note-card.css [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+/*tag container styling*/
+.tag-container {
+   width: 100%;
+   height: max-content;
+   padding: 0 1.5rem;
+
+   display: flex;
+   flex-wrap: wrap;
+   gap: 10px;
```
Comment (varun13169) [inline src/components/create-note-card/create-note-card.css:100]: Shouldn't the gap change with font size? use rem.

File: src/assets/icons.js (hunk 2 above)
Comment (amartya-rs) [reply src/assets/icons.js]: Good point, made the changes.
[changes](https://github.com/amartya-rs/dopeNotes/pull/3/commits/e89db64b47492b8a005747c827d4bf427521bb86)

File: src/assets/icons.js (hunk 3 above)
Comment (amartya-rs) [reply src/assets/icons.js]: Made the changes.

File: src/assets/icons.js (hunk 4 above)
Comment (amartya-rs) [reply src/assets/icons.js]: Made the changes.

File: src/components/tag-input/TagInput.jsx (hunk 1 above)
Comment (amartya-rs) [reply src/components/tag-input/TagInput.jsx:11]: Thanks for your suggestion. Will look into it.

File: src/components/color-pallete/color-palette.css (hunk 5 above)
Comment (amartya-rs) [reply src/components/color-pallete/color-palette.css:5]: Thanks for pointing this out. Will work on it.

File: src/components/create-note-card/create-note-card.css (hunk 6 above)
Comment (amartya-rs) [reply src/components/create-note-card/create-note-card.css:100]: Will work on it.

**Pull request merged**